zookeeper-server/src/test/java/org/apache/zookeeper/test/AuthSHA2Test.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import org.apache.zookeeper.server.auth.DigestAuthenticationProvider;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AuthSHA2Test extends AuthTest {
+
+    @BeforeEach
+    @Override
+    public void setup() {
+        // use the BouncyCastle's Provider for testing
+        Security.addProvider(new BouncyCastleProvider());
```

Comment:
this is going to pollute all of the other tests in an unpredictable way.
is there a way to remove it ?

Reply:
I don't think it will have any side-effect. (you can have as many security providers registered as many you wish... the JVM will go through them one-after-another in the order of their registrations, asking them for the given algorithm)

It still might be more elegant to register the provider in a @BeforeAll method. But doesn't really matter, as according to the JavaDocs it won't register the same security provider twice in the same JVM. (if I read it correctly: https://docs.oracle.com/javase/8/docs/api/java/security/Security.html#addProvider-java.security.Provider-)

Reply:
maybe I misunderstood your comment... thinking about side effects, maybe doing a 
```
System.clearProperty(DigestAuthenticationProvider.DIGEST_ALGORITHM_KEY);
System.clearProperty("zookeeper.authProvider.1");
```
in an `@AfterAll` method could be important here indeed to clean-up our custom configs after the tests.

Reply:
Good catch